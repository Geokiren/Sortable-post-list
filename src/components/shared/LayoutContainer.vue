<template>
  <div class="container" :class="classes">
    <slot></slot>
  </div>
</template>

<script setup lang="ts">
  defineProps<{
    classes?: string
  }>();
</script>