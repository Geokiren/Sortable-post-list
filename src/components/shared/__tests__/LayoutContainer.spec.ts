import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import LayoutContainer from '../LayoutContainer.vue';
import { DOMWrapper } from '@vue/test-utils';

describe('LayoutContainer', () => {
  it('Renders properly with classes through props', () => {
    const wrapper = mount(LayoutContainer, { props: { classes: 'bg-red' } });

    expect(wrapper.classes()).toContain('bg-red');
  });

  it('Renders properly with the correct element tag through slots', () => {
    const wrapper = mount(LayoutContainer, {
      slots: {
        default: '<div class="test">Test</div>'
      }
    });

    const slotElement: DOMWrapper<Element> = wrapper.find('.test');
    const tag = slotElement.element.tagName.toLowerCase();

    expect(tag).toBe('div');
  });

  it('Renders properly with the correct element text through slots', () => {
    const wrapper = mount(LayoutContainer, {
      slots: {
        default: '<div class="test">Test</div>'
      }
    });

    const slotElement: DOMWrapper<Element> = wrapper.find('.test');
    const text = slotElement.element.textContent;

    expect(text).toBe('Test');
  });

  it('Matches snapshot', () => {
    const wrapper = mount(LayoutContainer);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
