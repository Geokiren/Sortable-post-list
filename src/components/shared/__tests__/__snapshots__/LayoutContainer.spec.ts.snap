// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LayoutContainer > Matches snapshot 1`] = `"<div class="container"></div>"`;
